<launch>
	<arg name="resolution" default="1.0" />
	<arg name="compress_map" default="false" />

	<node pkg="octomap_server" type="octomap_server_node" name="octomap_server">
		<param name="resolution" value="$(arg resolution)" /> <!-- resolution 2 -->
		<param name="frame_id" type="string" value="world_enu" />
		<param name="sensor_model/max_range" value="40.0" />
		<param name="compress_map" value="$(arg compress_map)" />
		<remap from="cloud_in" to="/airsim_node/drone_1/lidar/LidarCustom" />\
	</node>

	<node pkg="octomap_server" type="octomap_server_node_trimmed" name="octomap_server_trimmed">
		<param name="resolution" value="$(arg resolution)" /> <!-- resolution 2 -->
		<param name="frame_id" type="string" value="world_enu" />
		<param name="sensor_model/max_range" value="40.0" />
		<param name="compress_map" value="$(arg compress_map)" />
		<remap from="cloud_in" to="/airsim_node/drone_1/lidar/LidarCustomTrimmed" />
	</node>

	<node pkg="octomap_server" type="octomap_server_node_highres" name="octomap_server_highres">
		<param name="resolution" value="0.1" /> <!-- resolution 2 -->
		<param name="frame_id" type="string" value="world_enu" />
		<param name="sensor_model/max_range" value="40.0" />
		<param name="compress_map" value="$(arg compress_map)" />
		<remap from="cloud_in" to="/airsim_node/drone_1/lidar/LidarCustom" />
	</node>

	<node pkg="octomap_server" type="airsim_zFilter" name="airsim_zFilter" output="screen">
		<param name="resolution" value="$(arg resolution)" />
		<param name="trimmed_height" value="0.6" />
	</node>

	<node pkg="octomap_server" type="airsim_visitedPoints_Publisher" name="airsim_visitedPoints_Publisher"></node>
</launch>
